### PR TITLE
clean up dynamic topography plugins

### PR DESCRIPTION
**Pull request by tjhei**
- simplify code
- suppress warnings about uninitialized variables

### CODE REVIEW

File: source/postprocess/dynamic_topography.cc [hunk 1]
```diff
@@ -149,15 +149,9 @@ namespace aspect
...
-              Point<dim> midpoint_at_surface;
-              for (unsigned int f = 0; f < GeometryInfo<dim>::faces_per_cell; ++f)
-                if (cell->at_boundary(f) && this->get_geometry_model().depth (cell->face(f)->center()) < cell->face(f)->minimum_vertex_distance()/3)
-                  {
-                    fe_face_values.reinit(cell,f);
-                    surface = fe_face_values.JxW(0);
-                    midpoint_at_surface = cell->face(f)->center();
-                  }
+              fe_face_values.reinit(cell, top_face_idx);
+              double surface = fe_face_values.JxW(0);
```
Comment (gassmoeller) [inline source/postprocess/dynamic_topography.cc]: Can't this be 'const double surface = ' ?

File: source/postprocess/visualization/dynamic_topography.cc [hunk 2]
```diff
@@ -155,13 +154,8 @@ namespace aspect
...
                 // Compute the associated surface area to later compute the surfaces weighted integral
-                double surface = 0;
-                for (unsigned int f = 0; f < GeometryInfo<dim>::faces_per_cell; ++f)
-                  if (cell->at_boundary(f) && this->get_geometry_model().depth (cell->face(f)->center()) < cell->face(f)->minimum_vertex_distance()/3)
-                    {
-                      fe_face_values.reinit(cell,f);
-                      surface = fe_face_values.JxW(0);
-                    }
+                fe_face_values.reinit(cell, top_face_idx);
+                double surface = fe_face_values.JxW(0);
```
Comment (gassmoeller) [inline source/postprocess/visualization/dynamic_topography.cc]: same here

**Comment by gassmoeller** [comment]
Nice improvements. It simplifies the logic and speeds up things a bit. 
And of course every pull request with more deletions than insertions is a good one (except you are removing functionality).

**Comment by tjhei** [comment]
sprinkled in some ```const```.

**Comment by gassmoeller** [comment]
Ok great. Tester is happy as well.

**Comment by bangerth** [comment]
I see that we trained Rene well -- I looked at the patch just before going to teach and mulled over the missing consts :-)

**Pull request merged**